testing/todocheck_test.go

```diff
@@ -397,6 +399,42 @@ func TestConfigDerivedFromBasepath(t *testing.T) {
 	}
 }
 
+func TestConfigAutoDetect(t *testing.T) {
+	// Since we're unable to have a .git directory anywhere inside the repository,
+	// make sure it's there beforehand.
+	gitDir := "./scenarios/auto_detect_config/.git"
+	gitConfig := `[remote "origin"]
+	url = git@github.com:test_username/test_repo.git
```

Comment:
Could you also add a test for detecting an origin specified without the `.git` extension? E.g.
```
	url = https://github.com/preslavmihaylov/todocheck
```

Reply:
This is why I requested the change about `WithGitConfig`

Reply:
Full configuration example:
```
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	precomposeunicode = true
[remote "origin"]
	url = https://github.com/preslavmihaylov/todocheck
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
	rebase = true
```

Reply:
Done.